metalearners/metalearner.py

```diff
@@ -1040,6 +1040,57 @@ def _default_scoring() -> Scoring:
             return default_scoring
         return dict(default_scoring) | dict(scoring)
 
+    def _validate_onnx_models(
+        self, models: Mapping[str, Sequence], necessary_models: Set[str]
+    ):
+        if set(models.keys()) != necessary_models:
+            raise ValueError(f"{necessary_models} should be present in models keys.")
+        specs_look_up = (
+            self.nuisance_model_specifications() | self.treatment_model_specifications()
+        )
+        input_format = None
+        for model_kind in necessary_models:
+            model_specs = specs_look_up[model_kind]
+            if len(models[model_kind]) != model_specs["cardinality"](self):
+                raise ValueError(
+                    f"{model_kind} cardinality does not match the expected cardinality."
+                )
+            predict_method = model_specs["predict_method"](self)
+            for i, m in enumerate(models[model_kind]):
+                if input_format is None:
+                    input_format = m.graph.input
+                elif input_format != m.graph.input:
+                    raise ValueError(
+                        "Some ONNX model has a different input, check that all models have "
+                        "the same input format."
+                    )
+                if predict_method == "predict" and len(m.graph.output) != 1:
+                    raise ValueError(
+                        f"ONNX {model_kind} with index {i} has {len(m.graph.output)} "
+                        "outputs and should have only one."
+                    )
+                elif predict_method == "predict_proba":
+                    found_probabilities = False
+                    for output in m.graph.output:
+                        if output.name in ["probabilities", "output_probability"]:
+                            found_probabilities = True
+                    if not found_probabilities:
+                        raise ValueError(
+                            f"ONNX {model_kind} with index {i} needs to have an output "
+                            "with name 'probabilities' or 'output_probability'."
+                        )
+
+    def _validate_feature_set_all(self):
+        for feature_set in self.feature_set.values():
+            if feature_set is not None:
```

Comment:
I think strictly speaking this condition is not equivalent to the error message. Couldn't it be that a user explicitly passes all available features as parameters to each `feature_set` entry?

Reply:
Good catch! I changed the error and method name:
[aa02226](https://github.com/Quantco/metalearners/pull/39/commits/aa02226704049b3e074fd8f3d6e37e210626044a)